module/dkg/broker.go

```diff
@@ -209,10 +261,18 @@ func (b *Broker) SubmitResult(pubKey crypto.PublicKey, groupKeys []crypto.Public
 	if err != nil {
 		b.log.Fatal().Err(err).Msg("failed to create retry mechanism")
 	}
-	maxedExpRetry := retry.WithMaxRetries(retryMax, expRetry)
+	maxedExpRetry := retry.WithMaxRetries(retryMaxPublish, expRetry)
 
+	attempts := 0
 	err = retry.Do(context.Background(), maxedExpRetry, func(ctx context.Context) error {
-		err := b.dkgContractClient.SubmitResult(pubKey, groupKeys)
+		attempts++
+		// retry with next fallback client after 2 failed attempts
+		if attempts%2 == 0 {
+			b.log.Warn().Msgf("submit result: retrying on attempt (%d) with fallback access node", attempts)
+			b.updateActiveDKGContractClient()
```

Comment:
Same as above: here I would actually call the update first, and then print the `activeQCContractClient` in the message, so that the engineer trying to debug this from logs will know which node is attempted when.

Reply:
Nit: missed the change from d8c0995 here